### PR TITLE
Migrates Continuous Deployment from Travis to Github Actions

### PR DESCRIPTION
**Pull request by ismail-shahzad**
Issue: https://gitlab.com/datopian/tech/devops/-/issues/367
Epic: https://gitlab.com/datopian/tech/devops/-/issues/366

The travis CI/CD is migrated to github actions.
The [`.travis.yml`](https://github.com/datopian/ckan-cloud-helm/blob/master/.travis.yml) file has been translated to [`main.yml`](https://github.com/datopian/ckan-cloud-helm/blob/travis_migration/.github/workflows/main.yml)

The pipeline only runs when a new tag is created.
A Deploy Key has been created which is then encrypted using AES algorithm. The AES key and IV are stored in two secret variables (`TRAVIS_KEY` and `TRAVIS_IV`) created in the repositroy as they are needed to decrypt the Deploy key. 
Also a few directory paths have been modified from the paths that were being used in Travis.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        mkdir -p $HOME/bin/
+        mv travis_ci_operator.sh $HOME/bin/travis_ci_operator.sh
+        sudo bash $HOME/bin/travis_ci_operator.sh init
+        sudo bash .travis.sh install
+    - name: Run the script
+      run: |
+        sudo bash .travis.sh script
+    - name: Deploy
+      env:
+        TRAVIS_REPO_SLUG: MuhammadIsmailShahzad/ckan-cloud-helm
```
Comment (ismail-shahzad) [inline .github/workflows/main.yml]: @zelima just caught my own mistake this should be the link to datopian/ckan-cloud-helm
Fixing now

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    services:
+      docker:
+        image: docker
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2.3.1
+    - name: Download and Install
+      uses: wei/curl@master
+      with:
+          args: curl -L https://raw.githubusercontent.com/datopian/ckan-cloud-helm/master/travis_ci_operator.sh > travis_ci_operator.sh
```
Comment (zelima) [inline .github/workflows/main.yml]: I don't get point of this if we add `travis_ci_operator.sh` file itself below

File: travis_ci_operator.sh [hunk 3]
```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env bash
```
Comment (zelima) [inline travis_ci_operator.sh:1]: @MuhammadIsmailShahzad So either we keep this or remove and get it with curl

**Review by zelima — APPROVED** [review]

File: travis_ci_operator.sh (hunk 3 above)
Comment (ismail-shahzad) [reply travis_ci_operator.sh:1]: @zelima 
Will keep this
Was using it with curl before
Added it in the repo in the final PR forgot about removing curl
Will remove the curl command

File: .github/workflows/main.yml (hunk 1 above)
Comment (zelima) [reply .github/workflows/main.yml]: :+1: eagle eye

**Comment by ismail-shahzad** [comment]
@zelima Ready for review

File: travis_ci_operator.sh (hunk 3 above)
Comment (zelima) [reply travis_ci_operator.sh:1]: But why removing curl?

Comment (zelima) [reply travis_ci_operator.sh:1]: What if smth changes there?

**Pull request merged**